direct/environment.py

```diff
@@ -216,13 +215,13 @@ def initialize_models_from_config(
     for k, v in cfg.additional_models.items():
         # Remove model_name key
         curr_model = models[k]
-        curr_model_cfg = {kk: vv for kk, vv in v.items() if kk != "model_name"}
+        curr_model_cfg = {kk: vv for kk, vv in v.items() if kk not in ["engine_name", "model_name"]}
         additional_models[k] = curr_model(**curr_model_cfg)
 
     model = models["model"](
         forward_operator=forward_operator,
         backward_operator=backward_operator,
-        **{k: v for (k, v) in cfg.model.items()},
+        **{k: v for (k, v) in cfg.model.items() if k != "engine_name"},
     ).to(device)
```

Comment:
I somehow feel this should be done differently, but I don't have a clear way to do it. Can't it happen right now that the model name is actually equal to model_name?

Reply:
If it's the same as model_name this is how it's defined (Line 269):
```engine_name = cfg.model.engine_name if cfg.model.engine_name else cfg.model.model_name.split(".")[-1] + "Engine"```

The rationale is that for the SSL/JSSL engines the same model is still used as the original Engine. So it wouldn't make sense to define new models just for new different Engines.